File: 220831/JavaScript/uniform.test.js
import {
  deduplicate, sortArrays, findStudentsCantLendClothes, solution,
} from './uniform';

test('Deduplicate Losts and Reserves', () => {
  const losts = [1, 2, 3, 5, 6, 7];
  const reserves = [3, 4, 5];

  const { uniqueLosts, uniqueReserves } = deduplicate(losts, reserves);

  expect(uniqueLosts).toStrictEqual([1, 2, 6, 7]);
  expect(uniqueReserves).toStrictEqual([4]);
});

test('Not Deduplicate Losts and Reserves', () => {
  const losts = [2, 4];
  const reserves = [1, 3];

  const { uniqueLosts, uniqueReserves } = deduplicate(losts, reserves);

  expect(uniqueLosts).toStrictEqual([2, 4]);
  expect(uniqueReserves).toStrictEqual([1, 3]);
});

test('Sort arrays', () => {
  const losts = [5, 13, 6, 22];
  const reserves = [35, 4, 9, 31, 3];

  const { sortedLosts, sortedReserves } = sortArrays(losts, reserves);

  expect(sortedLosts).toStrictEqual([5, 6, 13, 22]);
  expect(sortedReserves).toStrictEqual([3, 4, 9, 31, 35]);
});

test('Not Sort arrays', () => {
  const losts = [2, 4];
  const reserves = [1, 3];

  const { sortedLosts, sortedReserves } = sortArrays(losts, reserves);

  expect(sortedLosts).toStrictEqual([2, 4]);
  expect(sortedReserves).toStrictEqual([1, 3]);
});

test('Check if students who lost clothes can lend clothes to other students', () => {
  const lostsCase1 = [2, 4];
  const reservesCase1 = [1, 3];

  const studentsUnableToAttendCase1 = findStudentsCantLendClothes(lostsCase1, reservesCase1);

  expect(studentsUnableToAttendCase1).toStrictEqual([]);

  const lostsCase2 = [1, 2, 3];
  const reservesCase2 = [4, 5, 6, 7];

  const studentsUnableToAttendCase2 = findStudentsCantLendClothes(lostsCase2, reservesCase2);

  expect(studentsUnableToAttendCase2).toStrictEqual([1, 2]);
});

test('Check if students who lost clothes cannot lend clothes to other students', () => {
  const lostsCase1 = [1, 2, 3];
  const reservesCase1 = [5, 6, 7, 8];

  const studentsUnableToAttendCase1 = findStudentsCantLendClothes(lostsCase1, reservesCase1);

  expect(studentsUnableToAttendCase1).toStrictEqual([1, 2, 3]);
});

test('solution', () => {
  const studentsCountCase1 = 9;
  const lostsCase1 = [5, 6, 8, 1, 2];
  const reservesCase1 = [2, 3, 1, 4, 8, 9];

  expect(solution(studentsCountCase1, lostsCase1, reservesCase1)).toBe(8);

  const studentsCountCase2 = 7;
  const lostsCase2 = [2, 4];
  const reservesCase2 = [1, 3];

  expect(solution(studentsCountCase2, lostsCase2, reservesCase2)).toBe(7);
});

File: 220831/Java/UniformTest.java
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class UniformTest {
  @Test
  void solution() {
    Uniform test = new Uniform();

    assertEquals(
        6,
        test.solution(8, new int[]{1, 3, 4, 6, 7}, new int[]{2, 5, 6})
    );

    assertEquals(
        28,
        test.solution(
            30,
            new int[]{1, 2, 3},
            new int[]{4, 5, 6, 7}
        )
    );

    assertEquals(
        5,
        test.solution(5, new int[]{2, 4}, new int[]{1, 3, 5})
    );

    assertEquals(
        4,
        test.solution(5, new int[]{2, 4}, new int[]{3})
    );

    assertEquals(
        2,
        test.solution(3, new int[]{3}, new int[]{1})
    );
    
    assertEquals(
        4,
        test.solution(5, new int[]{1, 2}, new int[]{2, 3})
    );

    assertEquals(5,test.solution(5,new int[] {1,2},new int[] {1,2}));
    assertEquals(4,test.solution(5,new int[] {2,3,4},new int[] {3,5}));
    assertEquals(2,test.solution(3,new int[] {1,2},new int[] {2,3}));
    assertEquals(8,test.solution(9,new int[] {5,6,8,1,2},new int[] {2,3,1,4,8,9}));
    assertEquals(6,test.solution(6,new int[] {2,4,5},new int[] {1,3,6}));
    assertEquals(5,test.solution(5,new int[] {2,4,6},new int[] {1,3,5,7}));

    assertEquals(
        10,
        test.solution(
            10,
            new int[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
            new int[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}
        )
    );

    assertEquals(
        9,
        test.solution(
            10,
            new int[]{1, 2, 3, 4, 5, 6},
            new int[]{1, 2, 3, 4, 6, 7}
        )
    );

    assertEquals(
        6,
        test.solution(
            7,
            new int[]{2, 3, 5},
            new int[]{3, 4}
        )
    );

    assertEquals(
        6,
        test.solution(
            7,
            new int[]{2, 3, 5},
            new int[]{3, 4}
        )
    );

    assertEquals(
        1,
        test.solution(
            6,
            new int[]{1, 2, 3, 4, 5, 6},
            new int[]{1}
        )
    );

    assertEquals(
        6,
        test.solution(
            6,
            new int[]{1},
            new int[]{1, 2, 3, 4, 5, 6}
        )
    );

    assertEquals(
        1,
        test.solution(
            2,
            new int[]{1, 2},
            new int[]{1}
        )
    );

    assertEquals(
        3,
        test.solution(
            7,
            new int[]{1, 2, 3, 5, 6, 7},
            new int[]{3, 4, 5}
        )
    );

    assertEquals(
        3,
        test.solution(
            7,
            new int[]{1, 2, 3, 5, 6, 7},
            new int[]{3, 4, 5}
        )
    );
  }
}
